scheduler/src/cook/mesos/util.clj

```diff
@@ -438,22 +496,28 @@
 
 (defn task-ent->user
   [task-ent]
-  (get-in task-ent [:job/_instance :job/user]))
+  (letfn [( task-ent->user-miss
+            [task-ent]
+            (get-in task-ent [:job/_instance :job/user]))]
+    (lookup-cache-datomic-entity! task-user-cache task-ent->user-miss task-ent)))
 
 (def ^:const default-job-priority 50)
 
 
 (defn task->feature-vector
-  [task]
   "Vector of comparable features of a task.
    Last two elements are aribitary tie breakers.
    Use :db/id because they guarantee uniqueness for different entities
    (:db/id task) is not sufficient because synthetic task entities don't have :db/id
-    This assumes there are at most one synthetic task for a job, otherwise uniqueness invariant will break"
-  [(- (:job/priority (:job/_instance task) default-job-priority))
-   (:instance/start-time task (java.util.Date. Long/MAX_VALUE))
-   (:db/id task)
-   (:db/id (:job/_instance task))])
+   This assumes there are at most one synthetic task for a job, otherwise uniqueness invariant will break"
+  [task]
+  (letfn [(task->feature-vector-miss
+             [task]
+             [(- (:job/priority (:job/_instance task) default-job-priority))
+              (:instance/start-time task (java.util.Date. Long/MAX_VALUE))
+              (:db/id task)
+              (:db/id (:job/_instance task))])]
+          (lookup-cache-datomic-entity! feature-vector-cache task->feature-vector-miss task)))
```

Comment:
These might be synthetic task entities from `create-task-ent` which won't have a `:db/id` field. You could use some logic like `(or (:db/id task) (:db/id (:job/_instance task)))` to pull either the task ID if present or the job ID. Since it's a map, I don't think this will be doing any datomic reads. 

I think it's also valid as a key because that mirrors the logic in the feature vector which is being cached.